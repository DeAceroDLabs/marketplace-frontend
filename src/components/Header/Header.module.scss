@use "styles/colors";

.header {
  height: 109px;
  width: 100%;
  top: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  text-align: center;
  justify-content: center;
}

.color {
  background-color: colors.$color-background-03;
  .showSearchBar {
    width: 100%;
    display: flex;
    justify-content: center;
  }
}

.clean {
  background-color: transparent;
  position: absolute;

  .doNotShowSearchBar {
    display: none;
  }

}

