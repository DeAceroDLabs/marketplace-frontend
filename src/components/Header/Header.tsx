import { useLocation } from "react-router-dom";
import styles from "./Header.module.scss";
import SearchBar from "components/common/SearchBar";

const Header: React.FC = () => {
  const location = useLocation();
  const activePath = location.pathname;
  const pathIsHome = activePath === "/";
  const backgroundColor = pathIsHome ? "clean" : "color";
  return (
    <div className={`${styles.header} ${styles[backgroundColor]}`}>
      {!pathIsHome && (
        <div className={styles["show-search-bar"]}>
          <SearchBar />
        </div>
      )}
    </div>
  );
};

export default Header;
