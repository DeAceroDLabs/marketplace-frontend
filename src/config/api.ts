import { API, Category, Product, ZipCodeLocation } from "./api.types";
import axios from "axios";
import { USER_API } from "./user.api.types";

const fetchData = (url: string) => {
  async function fetchUrl() {
    const response = await fetch(url);
    const json = await response.json();
    return json;
  }
  return fetchUrl();
};

const postRequest = async (url: string, body: any) => {
  const response = await axios.post(USER_API.createUser, body);
  return response;
};

const fetchCategories = (): Promise<Category[]> => {
  return fetchData(API.categories);
};

const fetchProducts = (categoryId: number): Promise<Product[]> => {
  return fetchData(API.products(categoryId));
};

const getLocation = (zipCode: string): Promise<ZipCodeLocation> => {
  return fetchData(API.location(zipCode)).then((data) => {
    if (data["location"]) {
      const neighborhoodsObject = data["location"]["neighborhood"];
      const neighborhoods = Object.keys(neighborhoodsObject);
      const neighborhoodsArray = neighborhoods.map((neighborhood) => {
        return {
          value: neighborhood,
          label: neighborhoodsObject[neighborhood],
        };
      });
      data["location"]["neighborhood"] = neighborhoodsArray;
    }
    return data;
  });
};

const searchProducts = (query: string): Promise<any> => {
  const body = {
    q: query,
  };
  return postRequest(API.search, body);
};

export {
  postRequest,
  fetchCategories,
  fetchProducts,
  searchProducts,
  getLocation,
};
