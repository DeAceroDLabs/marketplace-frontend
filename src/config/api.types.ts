export const API = {
  categories: "categories",
  products: (categoryId: number) => `categories/${categoryId}/products`,
  search: "/search",
  location: (zipCode: string) => `locations/postalCode/${zipCode}`,
};

export type Category = {
  categoryId: number;
  categoryName: string;
  imageUrl: string;
};

export type Article = {
  articleId: string;
};

export type Product = {
  productId: number;
  productName: string;
  categoryName: string;
  categoryId: number;
  articles: Article[];
  categoryImageUrl: string;
};

export type ZipCodeLocation = {
  countryName: string;
  location: Location;
};

export type Location = {
  cityId: string;
  city: string;
  stateId: string;
  state: string;
  neighborhood: Neighborhood[];
};

export type Neighborhood = {
  value: string;
  label: string;
};
