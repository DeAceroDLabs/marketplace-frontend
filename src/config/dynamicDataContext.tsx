import React from "react";

const defaultForm = {
  dynamicData: {} as any,
  setDynamicData: (data: any) => {},
};

export const DynamicDataContext = React.createContext(defaultForm);

export const DynamicDataProvider = DynamicDataContext.Provider;

export default DynamicDataContext;
