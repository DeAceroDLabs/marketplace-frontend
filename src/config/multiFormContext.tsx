import { Form } from "forms/form.types";
import React from "react";

const defaultForm = {
  forms: [] as Form[],
  activeForm: {} as Form,
  setActiveForm: (form: Form) => {},
};

const MultiFormContext = React.createContext(defaultForm);

export const MultiFormProvider = MultiFormContext.Provider;

export default MultiFormContext;
