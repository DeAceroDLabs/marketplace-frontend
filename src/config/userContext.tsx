import React from "react";

const defaultUser = {
  username: "",
  setUser: (username: string) => {},
};

const UserContext = React.createContext(defaultUser);

export const UserProvider = UserContext.Provider;

export default UserContext;
