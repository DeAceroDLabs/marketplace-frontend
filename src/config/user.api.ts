import { postRequest } from "./api";
import { CreateUser, USER_API } from "./user.api.types";

const createUser = (user: CreateUser) => {
  return postRequest(USER_API.createUser, user);
};
export { createUser };
