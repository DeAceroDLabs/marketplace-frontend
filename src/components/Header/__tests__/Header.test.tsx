import { render, fireEvent, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Header from "../Header";
import { screen } from "@testing-library/react";
import { UserProvider } from "config/userContext";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Header", () => {
  const mockUSer = {
    username: "mock_user",
    setUser: jest.fn(),
  };
  const setup = (route: string[]) => {
    return render(
      <UserProvider value={mockUSer}>
        <MemoryRouter initialEntries={route}>
          <Header />
        </MemoryRouter>
      </UserProvider>
    );
  };

  it("renders Header without problem", () => {
    const view = setup(["/"]);
    expect(view).toMatchSnapshot();
  });

  it("renders Header with color being on a different page", () => {
    const view = setup(["/search/mock"]);
    expect(view).toMatchSnapshot();
  });

  it("does not display search bar on home screen", () => {
    setup(["/"]);
    const searchBar = screen.queryByPlaceholderText("¿Qué estás buscando?");
    expect(searchBar).not.toBeInTheDocument();
  });

  it("home button useNavigate", async () => {
    setup(["/"]);
    const homeButton = screen.getAllByRole("button")[0];
    fireEvent.click(homeButton);
    await waitFor(() => {
      expect(mockedUsedNavigate).toHaveBeenCalled();
    });
  });

  it("login button useNavigate", async () => {
    setup(["/"]);
    const loginButton = screen.getAllByRole("button")[1];
    fireEvent.click(loginButton);
    await waitFor(() => {
      expect(mockedUsedNavigate).toHaveBeenCalled();
    });
  });

  it("cart button opens tooltip and close it with close icon", async () => {
    setup(["/"]);
    const cartButton = screen.getAllByRole("button")[2];
    fireEvent.click(cartButton);
    const tooltip = screen.queryByText('¡No tienes artículos en tu carrito!');
    await waitFor(() => {
      expect(tooltip).toBeInTheDocument()
    });
    const closeButton = screen.getAllByRole("button")[3];

    fireEvent.click(closeButton);

    await waitFor(() => {
      expect(screen.queryByText('¡No tienes artículos en tu carrito!')).not.toBeInTheDocument();
    });
  });

  it("cart button opens tooltip and close it with addproductsbutton", async () => {
    setup(["/"]);
    const cartButton = screen.getAllByRole("button")[2];
    fireEvent.click(cartButton);
    const tooltip = screen.queryByText('¡No tienes artículos en tu carrito!');
    await waitFor(() => {
      expect(tooltip).toBeInTheDocument()
    });
    const addProductsButton = screen.getAllByRole("button")[4];

    fireEvent.click(addProductsButton);

    await waitFor(() => {
      expect(mockedUsedNavigate).toHaveBeenCalled();
      expect(screen.queryByText('¡No tienes artículos en tu carrito!')).not.toBeInTheDocument();
    });
  });

  it("click on home and close tooltip ", async () => {
    setup(["/"]);
    const homeButton = screen.getAllByRole("button")[0];

    fireEvent.click(homeButton);

    await waitFor(() => {
      expect(screen.queryByText('¡No tienes artículos en tu carrito!')).not.toBeInTheDocument();
    });
  });

});
