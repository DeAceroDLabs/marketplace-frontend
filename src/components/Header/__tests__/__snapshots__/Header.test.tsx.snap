// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header with color being on a different page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header color"
      >
        <div
          class="logo-container"
        >
          <a
            href="/"
          >
            <div
              class="logo"
              id="logo"
            >
              Logo
            </div>
          </a>
        </div>
        <div
          class="search-container"
        >
          <div
            class="search-bar"
          >
            <input
              placeholder="¿Qué estás buscando?"
              type="text"
              value=""
            />
            <button>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bq05nv-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="login-container"
        >
          <button
            class="primary"
          >
            <div
              class="person-icon"
              id="user-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-12jf5c-MuiSvgIcon-root"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <div>
              Hola, 
              <b>
                Ingresa
              </b>
            </div>
          </button>
        </div>
        <div
          class="shopping-cart-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18e385h-MuiSvgIcon-root"
            data-testid="ShoppingCartIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header color"
    >
      <div
        class="logo-container"
      >
        <a
          href="/"
        >
          <div
            class="logo"
            id="logo"
          >
            Logo
          </div>
        </a>
      </div>
      <div
        class="search-container"
      >
        <div
          class="search-bar"
        >
          <input
            placeholder="¿Qué estás buscando?"
            type="text"
            value=""
          />
          <button>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bq05nv-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="login-container"
      >
        <button
          class="primary"
        >
          <div
            class="person-icon"
            id="user-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-12jf5c-MuiSvgIcon-root"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div>
            Hola, 
            <b>
              Ingresa
            </b>
          </div>
        </button>
      </div>
      <div
        class="shopping-cart-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18e385h-MuiSvgIcon-root"
          data-testid="ShoppingCartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header renders Header without problem 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header clean"
      >
        <div
          class="logo-container"
        >
          <a
            href="/"
          >
            <div
              class="logo"
              id="logo"
            >
              Logo
            </div>
          </a>
        </div>
        <div
          class="login-container"
        >
          <button
            class="primary"
          >
            <div
              class="person-icon"
              id="user-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-12jf5c-MuiSvgIcon-root"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <div>
              Hola, 
              <b>
                Ingresa
              </b>
            </div>
          </button>
        </div>
        <div
          class="shopping-cart-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18e385h-MuiSvgIcon-root"
            data-testid="ShoppingCartIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header clean"
    >
      <div
        class="logo-container"
      >
        <a
          href="/"
        >
          <div
            class="logo"
            id="logo"
          >
            Logo
          </div>
        </a>
      </div>
      <div
        class="login-container"
      >
        <button
          class="primary"
        >
          <div
            class="person-icon"
            id="user-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-12jf5c-MuiSvgIcon-root"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div>
            Hola, 
            <b>
              Ingresa
            </b>
          </div>
        </button>
      </div>
      <div
        class="shopping-cart-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18e385h-MuiSvgIcon-root"
          data-testid="ShoppingCartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
